### PR TITLE
Remove /t/ from URL paths

### LINKED ISSUE #54: Remove /t/ from URLs
As requested by @niemeyer, remove the `/t/` from the URLs, and redirect URLs with `/t/` to URLs without.

### PR DESCRIPTION
**Pull request by nottrobin**
Fixes #54

- Match paths without the `/t`
- In the model, give document dictionaries a "path" property, which is the link without the /t/
- Use this property for redirecting / to the home page link
- Replace links in the navigation to links without the `/t`

QA
--

`./run`, and:

- go to http://0.0.0.0:8030/ and see that you're redirected to http://0.0.0.0:8030/snap-documentation/3781 (without the `/t`)
- go to e.g. http://0.0.0.0:8030/t/getting-started/3876, and check you're redirected to http://0.0.0.0:8030/getting-started/3876 (removing the `/t`)
- check 404s still work - e.g. http://0.0.0.0:8030/nothing/9999 should show a 404 page
- check all links in the navigation don't have `/t` in them
- check other redirects work - e.g. http://0.0.0.0:8030/build-snaps/moos -> http://0.0.0.0:8030/moos-applications/7820

### CODE REVIEW

File: webapp/models.py [hunk 1]
```diff
@@ -165,6 +173,7 @@ def _parse_document_topic(self, topic):
                 updated_datetime.replace(tzinfo=None)
             ),
             "forum_link": f"{self.base_url}/t/{topic['slug']}/{topic['id']}",
```
Comment (tbille) [inline webapp/models.py]: Maybe it could be remove here to avoid having redirections all the times

**Comment by tbille** [comment]
Do you need to remove the `/t` here also?

https://github.com/canonical-websites/docs.snapcraft.io/blob/c258bfb4e38ba40bcd44592bac279183bea232b9/webapp/models.py#L184-L186

File: webapp/models.py (hunk 1 above)
Comment (nottrobin) [reply webapp/models.py]: What do you mean? The forum link will still have the `/t/`. Discourse hasn't changed.

**Comment by nottrobin** [comment]
No, I don't need to remove the `/t/` there. That would break the API calls, as Discourse's URL structure hasn't changed.

**Comment by tbille** [comment]
@nottrobin this is me badly reading the code

And last example in your QA step is not redirecting well:
 http://0.0.0.0:8030/build-snaps/moos  -> http://0.0.0.0:8030/t/moos-applications/7820

**Comment by nottrobin** [comment]
@tbille are you sure you're not mistaken? Here's what I see:

``` bash
robin@xps:~/git/docs.snapcraft.io (remove-t)$ curl -I http://0.0.0.0:8030/build-snaps/moos
HTTP/1.1 302 FOUND
Server: gunicorn/19.9.0
Date: Tue, 16 Oct 2018 21:50:44 GMT
Connection: keep-alive
Content-Type: text/html; charset=utf-8
Content-Length: 253
Location: http://0.0.0.0:8030/moos-applications/7820
X-VCS-Revision: c258bfb4e38ba40bcd44592bac279183bea232b9
X-Request-Id: 194a63a0-cee0-4a2f-98c6-3835ee1d5a08
```

And this line in the diff seems pretty conclusive:

https://github.com/canonical-websites/docs.snapcraft.io/pull/56/files#diff-cc3c13a007cd6eafc8674d192141c8dfR13

Also when I visit it in the browser it ends up on http://0.0.0.0:8030/moos-applications/7820 as expected.

File: redirects.yaml [hunk 2]
```diff
@@ -1,54 +1,54 @@
...
+core/install-opensuse/?: /installing-snapd/6735
+core/install-openwrt/?: /installing-snapd/6735
+core/install-raspbian/?: /installing-snap-on-raspbian/6754
+core/install-solus/?: /installing-snap-on-solus/6803
+core/install-ubuntu/?: /installing-snap-on-ubuntu/6740
+core/install/?: /installing-snapd/6735
+core/interfaces/?: /interfaces/6154
+core/updates/?: /keeping-snaps-up-to-date/7022
+core/usage/?: /getting-started/3876
+reference/channels/?: /channels/551s
```
Comment (tbille) [inline redirects.yaml]:
```suggestion
reference/channels/?: /channels/551
```

File: webapp/models.py (hunk 1 above)
Comment (tbille) [reply webapp/models.py]: as replied before, this me being bad

**Pull request merged**